mesonbuild/interpreter/interpreterobjects.py

```diff
@@ -320,6 +320,14 @@ def set_quoted_method(self, args: T.Tuple[str, str], kwargs: 'kwargs.Configurati
     @typed_kwargs('configuration_data.set10', _CONF_DATA_SET_KWS)
     def set10_method(self, args: T.Tuple[str, T.Union[int, bool]], kwargs: 'kwargs.ConfigurationDataSet') -> None:
         self.__check_used()
+        if isinstance(args[1], int):
+            FeatureDeprecated.single_use('configuration_data.set10 with number',
```

Comment:
Don't use FeatureDeprecated for this. It should be unconditionally deprecated, not only when users set their `meson_version: '>=0.61.0'`

Just use 
```python
if isinstance(args[1], int):
    mlog.deprecation('configuration_data.set10 received an invalid numeric argument, when only bools '
                     'are permitted. This has unexpected results for numbers other than 0 or 1. This '
                     'will become a hard error in a future version of meson.',
                     location=self.interpreter.current_node)
```

Reply:
I thought we were trying to stop using `mlog.deprecation` in favor of `FeatureDeprecated`?

Reply:
Are we?

We use mlog.warning all the time for things that are always warnings, maybe that would be preferable?

In my mind, the purpose of mlog is for things that always need to be logged, whatever the reason, and Features are for things that depend on what version of meson you're trying to support.

I guess an argument could be made that "deprecation" implies versions where it is okay to use, and that therefore this *should* be an mlog.warning either way.